site-content/source/modules/ROOT/pages/development/dependencies.adoc

```diff
@@ -12,13 +12,44 @@ As Cassandra is an Apache product, all included libraries must follow
 Apache's https://www.apache.org/legal/resolved.html[software license
 requirements].
 
+=== Changes to dependency management in Cassandra 4.2
+
+Before Cassandra 4.2, dependencies were managed in `build.xml` and managed with
+Maven Ant Tasks. In Cassandra 4.2 and onwards, dependencies are managed in
+Maven POM templates in `.build/*-template.xml`. These templates are processed
+into valid Maven POMs with the `ant write-poms` task and copied to
+`build/*.pom`.
+
+The following POMs from `build.xml` have been migrated to templates:
+* `parent-pom` is now at `.build/parent-pom-template.xml`
+* `all-pom` is now at `.build/cassandra-deps-template.xml`
+* `build-deps-pom` is now at `.build/cassandra-build-deps-template.xml`
+
+To add new dependencies in Cassandra 4.2 onwards, add the dependency to
+`parent-pom-template` and `cassandra-deps-template`, and optionally
+`cassandra-build-deps-template` if the dependency is required for build only.
+See
+https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Management[the
+Maven docs] on how to reference dependencies in the parent POM from the child
+POMs.
+
+For dependency versions that need to be available in `build.xml` and
+`parent-pom-template`, specify the version as a property in `build.xml`, add it
+to the `ant write-poms` target, then add the property to `parent-pom-template`
+with the value of the template substitution.
+
+For more information on this change, see
+https://issues.apache.org/jira/browse/CASSANDRA-17750[CASSANDRA-17750].
+
```

Comment:
i'm going to nit-pick a bit here… forgive me…

but i'm curious whether folk think this page would be more readable if it contains two separate unadulterated sections:
- Dependency Management 4.2+
- Dependency Management before 4.2

For example, the `Changes to dependency management in Cassandra 4.2` section puts unnecessary cognitive load onto the reader if they just don't care about how it worked before 4.2

Reply:
I'm up for that - how's my last edit? This separates those two sections a bit more clearly. We can always edit based on feedback from more users as well.

Reply:
I like the latest changes, I would probably just put before 4.2 first and then 4.2. POM file types shouldn't be only at the end I think or we need some kind of anchor to them maybe, where we refer to changes to be done to them or what changed in different versions.
Otherwise LGTM, this is more around how to organize to make the flow easier to follow. 
Thanks for looking into that!

Reply:
In my opinion, the main audience for this doc (in the short term) is developers who are familiar with how to add dependencies in the pre-4.2 world, and are trying to figure out how to do things post-17750. In that scenario, I think it makes sense to increase visibility around the changes by having that section come first. Over time, this will change as 4.2 becomes more familiar to more folks.

Reply:
> In my opinion, the main audience for this doc (in the short term) is developers who are familiar with how to add dependencies in the pre-4.2 world, and are trying to figure out how to do things post-17750.

I would disagree with this statement :) because that method only existed for a year, ref [CASSANDRA-16391](https://issues.apache.org/jira/browse/CASSANDRA-16391). There was a much longer precedence before (checking jar and license files into `lib/`) I do suspect most developers are next to starting from scratch (we don't add deps that often).

Reply:
I am with Mick here. Also, I don't like the tribe knowledge which I am honestly even sure that it is not the case here. Not many people deal with the dependencies or do it regularly. Also, yes, good point, things changed last year. I prefer to be more noisy and things to be more documented than less.